include/mesh/mesh_function.h

```diff
@@ -146,27 +146,45 @@ class MeshFunction : public FunctionBase<Number>,
 
   /**
    * Computes values at coordinate \p p and for time \p time, which
-   * defaults to zero.
+   * defaults to zero, optionally restricting the point to the passed
+   * subdomain_ids. This is useful in cases where there are multiple
+   * dimensioned elements, for example.
    */
   void operator() (const Point& p,
                    const Real time,
                    DenseVector<Number>& output);
 
   /**
+   * Computes values at coordinate \p p and for time \p time,
+   * restricting the point to the passed subdomain_ids. This is useful in
+   * cases where there are multiple dimensioned elements, for example.
+   */
+  void operator() (const Point& p,
+                   const Real time,
+                   DenseVector<Number>& output,
+                   const std::set<subdomain_id_type>* subdomain_ids);
+
+  /**
    * Computes gradients at coordinate \p p and for time \p time, which
```

Comment:
The former is a virtual function overriding the `FunctionBase` (pure virtual) definition. I was trying to avoid redoing the FunctionBase here because that could spiral on us a bit. However, I debated which way to go and went the lazy route. If you'd prefer me not introducing the new overload, I'll go back and update `FunctionBase` as well. My plan there would be the same as here: Add the new overload and keep the old one with a `libmesh_deprecated()` inside and have it call the new one with NULL for the default. Otherwise, we'll break the API on everyone subclassing FunctionBase. But maybe there's another way to go about it?